src/OrchardCore.Modules/OrchardCore.Redis/HealthChecks/RedisHealthCheck.cs

```diff
@@ -0,0 +1,55 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace OrchardCore.Redis.HealthChecks;
+
+public class RedisHealthCheck : IHealthCheck
+{
+    private readonly IServiceProvider _serviceProvider;
+
+    public RedisHealthCheck(IServiceProvider serviceProvider)
+    {
+        _serviceProvider = serviceProvider;
+    }
+
+    public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+    {
+        try
+        {
+            var redisService = _serviceProvider.GetService<IRedisService>();
+            if (redisService == null)
+            {
+                return HealthCheckResult.Unhealthy();
```

Comment:
Add a description to each `Unhealthy()` that provides information about what's exactly the issue.